bot/seasons/evergreen/trivia_quiz.py

```diff
@@ -121,56 +118,120 @@ def load_questions() -> dict:
-                    hint_no = 0
-                    unanswered += 1
-                    await self.send_score(ctx.channel, self.player_dict[ctx.channel.id])
+
+                    hint_no = 0  # init hint_no = 0 so that 2 hints/time alerts can be sent for the new question.
+
+                    await self.send_score(ctx.channel, self.game_player_scores[ctx.channel.id])
                     await asyncio.sleep(2)
 
             else:
+                if self.game_status[ctx.channel.id] is False:
+                    break
+                # Reduce points by 25 for every hint/time alert that has been sent.
                 points = 100 - 25*hint_no
-                if msg.author in self.player_dict[ctx.channel.id]:
-                    self.player_dict[ctx.channel.id][msg.author] += points
+                if msg.author in self.game_player_scores[ctx.channel.id]:
+                    self.game_player_scores[ctx.channel.id][msg.author] += points
                 else:
-                    self.player_dict[ctx.channel.id][msg.author] = points
+                    self.game_player_scores[ctx.channel.id][msg.author] = points
+
+                # Also updating the overall scoreboard.
+                if msg.author in self.player_scores:
+                    self.player_scores[msg.author] += points
+                else:
+                    self.player_scores[msg.author] = points
+
                 hint_no = 0
-                unanswered = 0
-                await ctx.send(f"{msg.author.mention} got the correct answer :tada: {points} points for ya.")
+
+                await ctx.send(f"{msg.author.mention} got the correct answer :tada: {points} points!")
                 await self.send_answer(ctx.channel, question_dict)
-                await self.send_score(ctx.channel, self.player_dict[ctx.channel.id])
+                await self.send_score(ctx.channel, self.game_player_scores[ctx.channel.id])
                 await asyncio.sleep(2)
 
+    @staticmethod
+    def make_start_embed() -> discord.Embed:
+        """Generate a starting/introduction embed for the quiz."""
+        start_embed = discord.Embed(colour=discord.Colour.red())
+        start_embed.title = "Quiz game Starting!!"
+        start_embed.description = "Each game consists of 5 questions.\n"
+        start_embed.description += "**Rules :**\nNo cheating and have fun!"
+        start_embed.set_footer(
+            text="Points for each question reduces by 25 after 10s or after a hint. Total time is 30s per question"
+        )
+        return start_embed
+
+    @quiz_game.command(name="stop")
+    async def stop_quiz(self, ctx: commands.Context) -> None:
+        """Stop a quiz game if its running in the channel."""
+        if self.game_status[ctx.channel.id] is True:
+            await self.stop(ctx.author, ctx.channel)
+        else:
+            await ctx.send("No quiz running.")
+
+    async def stop(self, author: discord.Member, channel: discord.TextChannel) -> None:
```

Comment:
Why is this a separate function? If it is kept as part of the command, decorators for checks can also be used (though maybe they shouldn't be if it adds more complexity, especially given that there is a custom error message).

Reply:
I think I will put the Stop function code inside the stop quiz command.